ge/org.osate.ge/src/org/osate/ge/DiagramExporter.java

```diff
@@ -0,0 +1,296 @@
+import org.eclipse.graphiti.mm.algorithms.Rectangle;
+import org.eclipse.graphiti.mm.algorithms.styles.Point;
+import org.eclipse.graphiti.mm.pictograms.Connection;
+import org.eclipse.graphiti.mm.pictograms.ConnectionDecorator;
+import org.eclipse.graphiti.mm.pictograms.ContainerShape;
+import org.eclipse.graphiti.mm.pictograms.Diagram;
+import org.eclipse.graphiti.mm.pictograms.FreeFormConnection;
+import org.eclipse.graphiti.mm.pictograms.Shape;
+import org.eclipse.graphiti.platform.IDiagramBehavior;
+import org.eclipse.graphiti.services.Graphiti;
+import org.eclipse.graphiti.ui.services.GraphitiUi;
+import org.eclipse.jface.dialogs.MessageDialog;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.widgets.Display;
+import org.osate.ge.internal.diagram.runtime.AgeDiagram;
+import org.osate.ge.internal.diagram.runtime.DiagramSerialization;
+import org.osate.ge.internal.diagram.runtime.layout.DiagramElementLayoutUtil;
+import org.osate.ge.internal.diagram.runtime.layout.LayoutInfoProvider;
+import org.osate.ge.internal.graphiti.AgeDiagramTypeProvider;
+import org.osate.ge.internal.graphiti.AgeFeatureProvider;
+import org.osate.ge.internal.graphiti.diagram.ColoringProvider;
+import org.osate.ge.internal.graphiti.diagram.GraphitiAgeDiagram;
+import org.osate.ge.internal.services.ExtensionRegistryService;
+import org.osate.ge.internal.ui.editor.AgeDiagramBehavior;
+import org.osate.ge.internal.util.ProjectUtil;
+import org.osgi.framework.FrameworkUtil;
+
+public class DiagramExporter {
+	/**
+	 * This method receives an aadl diagram file and an output file.
+	 * The diagram contained in the aadl diagram file is written to
+	 * the output file as a png.
+	 *
+	 * @param  diagramFile the diagram file to be exported
+	 * @param  outputFile  the file the image will be written to
+	 */
+	public static void exportDiagramAsPng(final IFile diagramFile, final File outputFile) throws IOException {
+		// Image container bounds
+		final ContainerBounds containerBounds = new ContainerBounds();
+		final URI uri = URI.createPlatformResourceURI(diagramFile.getFullPath().toString(), true);
+		final IProject project = ProjectUtil.getProject(uri);
+
+		// Create diagram
+		final org.osate.ge.diagram.Diagram mmDiagram = DiagramSerialization.readMetaModelDiagram(uri);
+
+		final TransactionalEditingDomain editingDomain = TransactionalEditingDomain.Factory.INSTANCE
+				.createEditingDomain();
+		final DummyDiagramBehavior dummyDiagramBehavior = new DummyDiagramBehavior(editingDomain, project);
+		final DummyDiagramTypeProvider dummyDiagramTypeProvider = new DummyDiagramTypeProvider(dummyDiagramBehavior);
+
+		final AgeDiagram ageDiagram = DiagramSerialization.createAgeDiagram(project, mmDiagram,
+				getExtensionRegistryService(dummyDiagramTypeProvider.getClass()));
+
+		// Display warning if the diagram is stored with a newer version of the diagram file format.
+		if (mmDiagram.getFormatVersion() > DiagramSerialization.FORMAT_VERSION) {
```

Comment:
Do not display any dialog as part of the export process. Just ignore version mismatch. This particular message is not relevant since it is being exported and not opened for editing.